packages/browser/src/Session.ts

```diff
@@ -227,6 +227,27 @@ export class Session extends EventEmitter {
         this
       )
     );
+
+    // When a session is logged in, we want to track its ID in local storage to
+    // enable silent refresh.
+    this.on("login", () => {
+      // Store the current session ID specifically in 'localStorage' (i.e., not using
+      // any other storage mechanism), as we don't deem this information to be
+      // sensitive, and we want to ensure it survives a browser tab refresh.
+      window.localStorage.setItem(KEY_CURRENT_SESSION, this.info.sessionId);
+    });
+
+    // On logout, silent refresh should no longer be considered a viable option.
+    this.on("logout", () => {
+      window.localStorage.removeItem(KEY_CURRENT_SESSION);
+    });
+
+    // If an error happens, silent refresh should no longer be considered a viable option
+    // to prevent endless redirection, and session information should be cleared.
+    this.on(EVENTS.ERROR, async () => {
+      window.localStorage.removeItem(KEY_CURRENT_SESSION);
+      await this.logout();
```

Comment:
I wasn't sure in here if calling `logout()` was correct — I think we actually want a "logout" that doesn't come from user interaction or SDK API call, that way we can differentiate a logout from a session had an error?

Reply:
I think a consuming app may already distinguish between the two, by registering different callbacks to `onError` vs `onLogout`. However, it is true that calling `logout` here will also send the logout signal. I'm not sure if it is a wrong thing or not...

Reply:
I feel like it may be?

Reply:
a66d20282c4d4abab963a65a6449e6e364ede7ba should clean this up. The logout signal is now only sent in case of user-initiated logout, and all the logout cleanup is located in one place.